doc/copp/CoPP Config and Management.md

```diff
@@ -0,0 +1,214 @@
+
+```
+    function postStartAction()
+        ...
+        if [ -r /etc/sonic/config_db.json ]; then
+            sonic-cfggen -j /etc/sonic/config_db.json --write-to-db
+            if [ -r /etc/sonic/init_cfg.json ]; then
+                sonic-cfggen -j /etc/sonic/init_cfg.json -j /etc/sonic/config_db.json --write-to-db
+            else
+                sonic-cfggen -j /etc/sonic/config_db.json --write-to-db
+            fi
+        fi
+```
+
+## Warmboot and Backward Compatibility
+It is desirable to have warmboot functionality from previous release versions of Sonic. Since the existing schema has COPP Group name/key with protocol names (e.g `"COPP_TABLE:trap.group.bgp.lacp"`, there is a limitation in adding any new protocol or trap to an existing CoPP group and seamlessly migrate. However, with this proposal, the implementation is to do a migration of APP DB entries to new schema.
+
+In addition, the implementation must ensure that backward compatibility is maintained. If the system is boot-up with an *old* config file, the default CoPP tables are to be loaded from ```init_cfg.json``` and expected to work seamlessly.
+
+## Limitations
+1. In case of downgrade, the config_db entries shall be present as stale entries as there is no subscribers for the table. Functionality would be same as supported by the downgraded version
+2. This proposal expects the table names to be consistent across multiple releases. 
+3. User is expected to resolve any conflicts, say for a trap id or group, that arises due to default values from the Sonic binary vs same trap or group currently or previously configured by the user.
+
+## CLI
+`show` commands to display CoPP group and CoPP entries shall be provided as part of this feature implementation.
+
+CLI support to add/modify CoPP tables is not scoped as part of this design. In future, the following commands shall be supported:
+
+1. User shall be able to change policer values for a queue
+2. User shall be able to change the queue for a protocol/trap
+3. User shall be able to delete a group/trap. 
+    *In the current proposal, this would mean to keep the key in config_db with empty attributes.
+
+# Flows
+
+The following flow diagram captures the control flow.
+
+## Initial config
+
+The following flow captures scenarios for ```boot-up``` sequence and ```config reload```. Default CoPP Tables shall be present in ```init_cfg.json``` and if the same entry is present in ```config_db.json```, it is expected to be overwritten by ```config_db``` entry. This model ensures user-configuration gets priority over default configuration.
+
+![](https://github.com/Azure/SONiC/blob/master/images/copp/copp_init.png)
+
+## Copp Manager flow
+
+The following flow captures CoPP manager functionality. 
+
+![](https://github.com/Azure/SONiC/blob/master/images/copp/CoppManager_1.png)
+
+# Examples
```

Comment:
It would help clarify the CoPPMgr behavior if you had an equivalent AppDB configuration next to this new ConfigDB version.

Reply:
Sure, added to examples section